pkgs/development/libraries/folly/default.nix

```diff
@@ -1,18 +1,37 @@
-{ stdenv, fetchFromGitHub, cmake, boost, libevent, double-conversion, glog
-, gflags, libiberty, openssl }:
+{ stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, libevent
+, double-conversion
+, glog
+, gflags
+, libiberty
+, lz4
+, lzma
+, zlib
+, jemalloc
+, openssl
+, pkg-config
+, libunwind
+, fmt
+}:
 
 stdenv.mkDerivation rec {
   pname = "folly";
-  version = "2019.11.11.00";
+  version = "2020.09.28.00";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "folly";
     rev = "v${version}";
-    sha256 = "1sgv7sdalbs7zhz3zcc95gn2h8j2xjf7hkw2c618zc3pdn6aa58w";
+    sha256 = "1ry2nqfavcbz0jvsqw71105gbxm5hpmdi2k1w155m957jrv3n5vg";
   };
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
 
```

Comment:
would you mind adding the following to fix the darwin build?
```
cmakeFlags = stdenv.lib.optionals stdenv.cc.isClang [ "-DCOMPILER_HAS_F_ALIGNED_NEW=OFF" ];
```

Reply:
Hmm, now I'm getting this:
```
  /private/var/folders/78/nbjn_b991sd60rhqbs_0xwd8x2nf6n/T/nix-build-folly-2020.09.28.00.drv-0/source/folly/concurrency/UnboundedQueue.h:561:21: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
  /private/var/folders/78/nbjn_b991sd60rhqbs_0xwd8x2nf6n/T/nix-build-folly-2020.09.28.00.drv-0/source/folly/concurrency/UnboundedQueue.h:565:7: error: aligned deallocation function of type 'void (void *, enum std::align_val_t) noexcept' is only available on macOS 10.13 or newer
        delete next;
        ^
  /private/var/folders/78/nbjn_b991sd60rhqbs_0xwd8x2nf6n/T/nix-build-folly-2020.09.28.00.drv-0/source/folly/concurrency/UnboundedQueue.h:565:7: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
  12 warnings and 18 errors generated.
```

Reply:
I cannot test on OSX, so I cannot help here. If someone finds a fix I'm happy to push that.

Reply:
If this does require the 10.13 SDK, then this will break on macos until the sdk used by nixpkgs gets bumped, which could potentially be a while. So I would think that we'd either need to block this update until that happens, or keep the old 2019.11.11.00 around for use on macs.
However I'm very new to nixpkgs so don't take this as the gospel truth. I've noticed a couple other packages failing on hydra due to requiring the 10.13 sdk so maybe this isn't a blocker.

Reply:
Was the 2019.11.11.00 build working on Mac (it wasn't for Linux)?

Reply:
On current nixpkgs master [the old version builds](https://hydra.nixos.org/build/128554246) for darwin.